scheduler/src/scheduling.test.ts

```diff
@@ -1,121 +1,113 @@
 /* eslint-disable @typescript-eslint/consistent-type-assertions */
-const mockedGetLatestEventId = getLatestEventId as jest.Mock
-mockedGetLatestEventId.mockResolvedValue(321)
 const mockedGetAllDatasources = getAllDatasources as jest.Mock
-const mockedGetEventsAfter = getEventsAfter as jest.Mock
-const mockedGetDatasource = getDatasource as jest.Mock
+const mockedTriggerDatasource = triggerDatasource as jest.Mock
 /* eslint-enable @typescript-eslint/consistent-type-assertions */
 
+const CONNECTION_RETRIES = 2
+const CONNECTION_BACKOFF_IN_MS = 1000
+const TRIGGER_RETRIES = 3
+
 jest.mock('./env', () => () => ({
   MAX_TRIGGER_RETRIES: 2
 }))
+mockedTriggerDatasource.mockReturnValue(Promise.resolve())
+
+let scheduler: Scheduler
 
 describe('Scheduler', () => {
+  beforeEach(async () => {
+    scheduler = new Scheduler(TRIGGER_RETRIES)
+  })
+
+  afterEach(async () => {
+    mockedTriggerDatasource.mockClear()
+  })
+
   test('should initialize jobs correctly', async () => {
-    const config = generateConfig(true, new Date(Date.now() + 5000), 6000)
+    const config = generateConfig(true, new Date(Date.now() + 500), 6000)
     mockedGetAllDatasources.mockResolvedValue([config])
+    await scheduler.initializeJobsWithRetry(CONNECTION_RETRIES, CONNECTION_BACKOFF_IN_MS)
 
-    await Scheduling.initializeJobsWithRetry()
+    await sleep(1000)
 
-    expect(Scheduling.getAllJobs()).toHaveLength(1)
-    expect(Scheduling.getAllJobs()[0].datasourceConfig).toEqual(config)
+    expect(scheduler.getAllJobs()).toHaveLength(1)
+    expect(scheduler.getAllJobs()[0].datasourceConfig).toEqual(config)
+    expect(mockedGetAllDatasources).toHaveBeenCalledTimes(1)
+    expect(mockedTriggerDatasource).toHaveBeenCalledTimes(1)
   })
 
   test('should apply creation event', async () => {
     mockedGetAllDatasources.mockResolvedValue([])
-    const toBeAdded = generateConfig(true, new Date(Date.now() + 5000), 6000)
-    const creationEvent = {
-      eventId: 322,
-      eventType: EventType.DATASOURCE_CREATE,
-      datasourceId: 123
+    const toBeAdded = generateConfig(true, new Date(Date.now() + 500), 6000)
+    const creationEvent: DatasourceConfigEvent = {
+      datasource: toBeAdded
     }
 
-    await Scheduling.initializeJobsWithRetry()
-    expect(Scheduling.getAllJobs()).toHaveLength(0)
-
-    mockedGetEventsAfter.mockResolvedValue([creationEvent])
-    mockedGetDatasource.mockResolvedValue(toBeAdded)
-    await Scheduling.updateDatasources()
+    await scheduler.applyCreateOrUpdateEvent(creationEvent)
 
-    await sleep(1000) // Give the scheduler some time to apply the update
+    await sleep(500)
 
-    expect(Scheduling.getAllJobs()).toHaveLength(1)
-    const job123 = Scheduling.getJob(123)
+    expect(scheduler.getAllJobs()).toHaveLength(1)
+    const job123 = scheduler.getJob(123)
     expect(job123).toBeDefined()
-    if (job123 !== undefined) {
-      expect(job123.datasourceConfig).toEqual(toBeAdded)
-    }
+    // @ts-expect-error
```

Comment:
I looked again at this problem. It seams like Optional Chaining is the better solution instead of the suppression of the TypeScript warning.

```js
    expect(job123?.datasourceConfig).toEqual(toBeAdded)
    // if job123 or the attribute datasourceConfig is not defined it will return 'undefined'
```